File: Laboratorio 3/Lab03parte1/app/src/main/java/com/example/lab03parte1/MainActivity.java
package com.example.lab03parte1;

import android.content.Intent;
import android.content.IntentFilter;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    private BatteryReceiver batteryReceiver;
    private TextView batteryStatusTextView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        batteryStatusTextView = findViewById(R.id.batteryStatusTextView);
        batteryReceiver = new BatteryReceiver(batteryStatusTextView);
    }

    @Override
    protected void onResume() {
        super.onResume();
        IntentFilter filter = new IntentFilter(Intent.ACTION_BATTERY_CHANGED);
        registerReceiver(batteryReceiver, filter);
        Log.d("MainActivity", "BatteryReceiver registro satisfactoriamente");
    }

    @Override
    protected void onPause() {
        super.onPause();
        unregisterReceiver(batteryReceiver);
        Log.d("MainActivity", "BatteryReceiver desregistro satisfactoriamente");
    }
}

File: Laboratorio1/app/src/main/java/com/example/laboratorio1/MainActivity.kt
package com.example.laboratorio1

import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import java.io.BufferedReader
import java.io.FileNotFoundException
import java.io.IOException
import java.io.InputStreamReader

class MainActivity : AppCompatActivity() {

    private val fileName = "registro.txt"

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

        var nombreEditText = findViewById<EditText>(R.id.edNombre)
        var apellidosEditText = findViewById<EditText>(R.id.edApellido)
        var correoEditText = findViewById<EditText>(R.id.edCorreo)
        var telefonoEditText = findViewById<EditText>(R.id.edTelefono)
        var grupoSanguineoEditText = findViewById<EditText>(R.id.edGrupoS)

        var btnRegistrar = findViewById<Button>(R.id.btnRegistrar)
        var btnMostrar = findViewById<Button>(R.id.btnMostrar)

        btnRegistrar.setOnClickListener {
            val nombre = nombreEditText.text.toString()
            val apellidos = apellidosEditText.text.toString()
            val correo = correoEditText.text.toString()
            val telefono = telefonoEditText.text.toString()
            val grupoSanguineo = grupoSanguineoEditText.text.toString()

            try {
                val fileOutputStream = openFileOutput(fileName, MODE_APPEND)
                val data = "$nombre, $apellidos, $correo, $telefono, $grupoSanguineo\n"
                fileOutputStream.write(data.toByteArray())
                fileOutputStream.close()
                Log.d("Info", "Datos guardados correctamente")
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }

        btnMostrar.setOnClickListener {
            try {
                val fileInputStream = openFileInput(fileName)
                val inputStreamReader = InputStreamReader(fileInputStream)
                val bufferedReader = BufferedReader(inputStreamReader)
                var line: String?

                while (bufferedReader.readLine().also { line = it } != null) {
                    Log.i("Registro", line!!)
                }

                bufferedReader.close()
            } catch (e: FileNotFoundException) {
                Log.e("Error", "Archivo no encontrado")
            } catch (e: IOException) {
                e.printStackTrace()
            }
        }
    }
}